browser/misc_metrics/page_metrics_tab_helper.cc

```diff
@@ -45,6 +68,56 @@ void PageMetricsTabHelper::DidFinishNavigation(
     is_reload = true;
   }
   page_metrics_->IncrementPagesLoadedCount(is_reload);
+
+  if (IsHttpAllowedForHost(navigation_handle) && was_http_allowlist_ &&
+      !navigation_handle->IsErrorPage()) {
+    page_metrics_->RecordAllowedHTTPRequest();
+  }
+}
+
+bool PageMetricsTabHelper::CheckNavigationEvent(
+    content::NavigationHandle* navigation_handle,
+    bool is_finished) {
+  if (!page_metrics_) {
+    return false;
+  }
+  if (!navigation_handle->IsInMainFrame() ||
```

Comment:
pls look at `IsInPrimaryMainFrame`, maybe we want this one?

Reply:
aslo we probably want to look at `IsInOutermostMainFrame`

Reply:
thanks. `IsInPrimaryMainFrame` seems like the right choice, especially considering that there could be multiple frame trees present.